app/models/manageiq/providers/azure/refresh_helper_methods.rb

```diff
@@ -57,4 +57,70 @@ def network_interfaces
   def ip_addresses
     @ip_addresses ||= gather_data_for_this_region(@ips)
   end
+
+  # Create the necessary service classes and lock down their api-version
+  # strings using the config/settings.yml from the provider repo. The
+  # "to_s" call for the version strings puts the date in the format
+  # that we need, i.e. "YYYY-MM-DD".
+  #
+
+  def availability_set_service(config)
+    @avs ||= ::Azure::Armrest::AvailabilitySetService.new(config)
```

Comment:
The config has a default api-version, but individual services have their own api-version strings. The config value is only used as a fallback, since we don't really want its value applied globally.

I always forget about tap/do. Thanks, I'll update that.

As for memoized vars, I suppose that could be an issue for multiple Azure providers, but I need to double check.